### PR TITLE
[RUNTIME] Cleanup build for libbacktrace

### PR DESCRIPTION
**Pull request by tqchen**
- Introduce TVM_USE_LIBBACKTRACE value macro to be consistent
  wth other value macros(instead of relying on disabled flag).
- Introduce AUTO mode for libbacktrace
- Temporary disable MacOS support in light of recent bug report.
- Refactor out the libbacktrace.cmake to libs
- Properly use TVM_DLL so that code is cross platform.
- Fallback to the weaker dmlc impl when backtrace is disabled.x

### CODE REVIEW

**Comment by tqchen** [comment]
cc @tkonolige @rkimball @junrushao1994

File: src/runtime/logging.cc [hunk 1]
```diff
@@ -17,16 +17,15 @@
...
 
-// TODO(bkimball,tkonolige) This inline function is to work around a linking error I am having when
-// using MSVC If the function definition is in logging.cc then the linker can't find it no matter
-// what kind of attributes (dllexport) I decorate it with. This is temporary and will be addressed
-// when we get backtrace working on Windows.
 namespace tvm {
 namespace runtime {
-__declspec(dllexport) std::string Backtrace() { return ""; }
+// Fallback to the dmlc implementation when backtrace is not available.
+std::string Backtrace() { return dmlc::StackTrace(); }
```
Comment (tkonolige) [inline src/runtime/logging.cc:28]: I don't think we should fall back to dmlc's stack trace implementation. It uses its own set of `-D` flags to control behavior. Users will be confused by the need to set `-DTVM_USE_LIBBACKTRACE=0` and `-DDMLC_LOG_STACK_TRACE=0`.

File: src/runtime/logging.cc [hunk 2]
```diff
@@ -17,16 +17,15 @@
  * under the License.
  */
 
-#ifdef TVM_BACKTRACE_DISABLED
+#if TVM_USE_LIBBACKTRACE == 0
```
Comment (tkonolige) [inline src/runtime/logging.cc:20]: I'd like to be consistent with other TVM #if usages.
```suggestion
#if TVM_USE_LIBBACKTRACE
```

**Review by tkonolige — CHANGES_REQUESTED** [review]
I think #7705 is caused by the user being on an M1 Mac. Let's keep backtraces enabled on x86_64 Macs.

File: src/runtime/logging.cc (hunk 1 above)
Comment (tqchen) [reply src/runtime/logging.cc:28]: the dmlc::StackTrace at least gives us something over nothing, so it is useful for systems when the backtrace is not enabled.

Comment (tkonolige) [reply src/runtime/logging.cc:28]: Can you move it into TVM then, so we have consistent flags.

Comment (tqchen) [reply src/runtime/logging.cc]: Given the fallback is less frequently used and this PR did fixes the previous problem and bring in new changes, I think this change is OK as it is. We can consider bring the new impl back later if needed.

**Comment by tqchen** [comment]
we have yet to confirm the problem in #7705 , once it is confirmed, feel free to send a new PR to enable MacOS x64

**Review by junrushao1994 — APPROVED** [review]

**Pull request merged**